Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/NamespaceSynchronizationConfig.swift

```diff
@@ -11,134 +11,67 @@ import StitchCoreSDK
  always be in sync.
  */
-internal class NamespaceSynchronization: Sequence {
-    /// The actual configuration to be persisted for this namespace.
-    class Config: Codable {
-        fileprivate enum CodingKeys: CodingKey {
-            case syncedDocuments, namespace
-        }
-        /// the namespace for this config
-        let namespace: MongoNamespace
-        /// a map of documents synchronized on this namespace, keyed on their documentIds
-        fileprivate(set) internal var syncedDocuments: [HashableBSONValue: CoreDocumentSynchronization.Config]
-        /// The conflict handler configured to this namespace.
-        fileprivate var conflictHandler: AnyConflictHandler?
-        /// The change event listener configured to this namespace.
-        fileprivate var changeEventDelegate: AnyChangeEventDelegate?
-
-        init(namespace: MongoNamespace,
-             syncedDocuments: [HashableBSONValue: CoreDocumentSynchronization.Config]) {
-            self.namespace = namespace
-            self.syncedDocuments = syncedDocuments
-        }
+final class NamespaceSynchronization: Sequence, Codable {
+    static func filter(namespace: MongoNamespace) -> Document {
+        return ["namespace": try? BSONEncoder().encode(namespace)]
     }
 
-    typealias Element = CoreDocumentSynchronization
-    typealias Iterator = NamespaceSynchronizationIterator
-
-    /// Allows for the iteration of the document configs contained in this instance.
-    struct NamespaceSynchronizationIterator: IteratorProtocol {
-        typealias Element = CoreDocumentSynchronization
-        private typealias Values = Dictionary<HashableBSONValue, CoreDocumentSynchronization.Config>.Values
-
-        private let docsColl: ThreadSafeMongoCollection<CoreDocumentSynchronization.Config>
-        private var values: Values
-        private var indices: DefaultIndices<Values>
-        private weak var errorListener: FatalErrorListener?
-
-        init(docsColl: ThreadSafeMongoCollection<CoreDocumentSynchronization.Config>,
-             values: Dictionary<HashableBSONValue, CoreDocumentSynchronization.Config>.Values,
-             errorListener: FatalErrorListener?) {
-            self.docsColl = docsColl
-            self.values = values
-            self.indices = self.values.indices
-            self.errorListener = errorListener
-        }
-
-        mutating func next() -> CoreDocumentSynchronization? {
-            guard let index = self.indices.popFirst() else {
-                return nil
-            }
-
-            return try? CoreDocumentSynchronization.init(docsColl: docsColl,
-                                                         config: &values[index],
-                                                         errorListener: errorListener)
-        }
+    enum CodingKeys: String, CodingKey {
+        case namespace = "namespace", schemaVersion = "schema_version", docsColl = "docs_coll"
```

Comment:
[task] add a note to https://docs.google.com/document/d/1vaeI7m8Gv7zWDh0yETfqgX3JOeiX88k2Y9STCwsAKp0/edit?usp=drive_web&ouid=101429542541116054715 indicating that this is different from the on-device representation on Android. There is no `docs_coll` field on Android. Whether or not we want this is related to the other discussion.